src/workshops/colors/colorPickerView.ts

```diff
@@ -0,0 +1,40 @@
+import * as ko from "knockout";
+import template from "./colorPickerView.html";
+import { Component, Param, Event, OnMounted } from "@paperbits/common/ko/decorators";
+import { ColorContract } from "../../contracts/colorContract";
+
+@Component({
+    selector: "color-picker-view",
+    template: template
+})
+export class ColorPickerView {
+
+    private selectTimeout: any;
+
+    @Param()
+    public readonly colorValue: ko.Observable<string>;
+
+    @Event()
+    public readonly onSelect: (colorValue: string) => void;
+
+    constructor() {
+
+        this.colorValue = ko.observable();
+    }
+
+    @OnMounted()
+    public async loadColors(): Promise<void> {
+        this.colorValue.subscribe(this.onColorChange);
+    }
+
+    public onColorChange(): void {
+        clearTimeout(this.selectTimeout);
+        this.selectTimeout = setTimeout(() => this.scheduleColorUpdate(), 50);
```

Comment:
This is an old approach, it turned out that there is a standard rate-limit extension in KO. Please have a look at pages.ts, there is a snippet:
```
    @OnMounted()
    public async initialize(): Promise<void> {
        await this.searchPages();

        this.searchPattern
            .extend(ChangeRateLimit)
            .subscribe(this.searchPages);
    }
```